src/components/popover-container/popover-container.component.tsx

```diff
@@ -168,6 +169,33 @@ export const PopoverContainer = ({
       setTimeout(() => closeButtonRef.current?.focus(), 0);
   }, [isOpen]);
 
+  const handleEscKey = useCallback(
+    (ev) => {
+      const eventIsFromSelectInput = Events.composedPath(ev).find((p) => {
```

Comment:
**comment**: I'm not sure how to fix it but I just tested this with the "complex example" story you kindly added the Select to, and notice that with that the Escape key doesn't close the PopoverContainer whenever the select is focused, even if it's shut. I assume the intended behaviour is that when the select list is open, escape only closes that, but that a subsequent escape press should close the whole thing? As it is you need to tab/click off the select for that to work. Not critical perhaps but something to think about.

Reply:
I think I've managed to address this using an `aria-expanded` DOM query.